scripts/mbedtls_framework/c_wrapper_generator.py

```diff
@@ -65,42 +67,50 @@ def _write_prologue(self, out: typing_util.Writable, header: bool) -> None:
 
         This includes a description comment and some include directives.
         """
-        out.write("""/* Automatically generated by {}, do not edit! */
+        prologue = textwrap.dedent("""
+            /* Automatically generated by generate_psa_wrappers.py, do not edit! */
+
+            /* Copyright The Mbed TLS Contributors
+             * SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+             */
+
+        """).lstrip()
```

Comment:
> Also, why complicate things with lstrip?

That allows more flexibility on how the strings are provided.

```
a = """ /* Automatically generated by generate_psa_wrappers.py, do not edit! */

           /* Copyright The Mbed TLS Contributors
""""
```

will produce the same result as:

```
a = """
            /* Automatically generated by generate_psa_wrappers.py, do not edit! */

            /* Copyright The Mbed TLS Contributors
"""
```

In more cases we may want to add empty newlines by appending at the end of the string, but it is not very usefull to append at the beggining of it.

> Also, I don't like using dedent: it only works if at least one line is indented at column 0, and it makes the generated snippets harder to read, especially if we want to use 80 columns of text

Personal preference aside,  [PEP style guide](https://peps.python.org/pep-0008/) recommends implicit concatenation as the way of handling multiline string lines. 

> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.
textwrap.dedent and inspect.cleandoc has been consis

For those who need other functionality textwrapt module has been the canonical way of handling it. Using mutlinline docutils breaking class identation hyerarcy is a non-cannonical approach, and we want the code to be maintained by people who are not familiar with it.

The limitations of dedent, do not apply in our use case since we are adding code at line 0. If something special is needed we can extend that function, but that is out of scope right now.

I will be adding a `blockstr_format` utiility function so we can update only that segment of code if needed.

Reply:
> That allows more flexibility on how the strings are provided.

I don't see how that's a good thing. It allows more flexibility to make the source code less readable. Rather a bad thing in my view.

> PEP style guide recommends implicit concatenation as the way of handling multiline string lines.

That's about long lines of Python code. PEP 8 doesn't say anything about long string literals.

> For those who need other functionality textwrapt module has been the canonical way of handling it.

I don't know what other functionality you're referring to, but `textwrap` is mostly about text wrapping and we don't want text wrapping here.

> Using mutlinline docutils breaking class identation hyerarcy is a non-cannonical approach, and we want the code to be maintained by people who are not familiar with it.

I don't know what you would consider to be the canonical approach here. And I don't see what's hard to maintain about having string literals that are used as presented in the source code, as opposed to being munged by some function that readers may be unfamiliar with.

> The limitations of dedent, do not apply in our use case since we are adding code at line 0.

ITYM at column 0. We are not only adding code at column 0: we often add code inside functions.

So my objection still stands: you're making the code less readable.